src/sentry/static/sentry/app/components/events/relatedEvents/index.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import capitalize from 'lodash/capitalize';
+import {Location} from 'history';
+
+import space from 'app/styles/space';
+import {t} from 'app/locale';
+import DateTime from 'app/components/dateTime';
+import {Organization} from 'app/types';
+import EventView from 'app/utils/discover/eventView';
+import {getTransactionDetailsUrl} from 'app/views/performance/utils';
+import {PanelTable, PanelItem} from 'app/components/panels';
+import PlatformIcon from 'app/components/platformIcon';
+import Link from 'app/components/links/link';
+import {TableDataRow} from 'app/utils/discover/discoverQuery';
+import Tooltip from 'app/components/tooltip';
+import {generateEventSlug, eventDetailsRouteWithEventView} from 'app/utils/discover/urls';
+import {IconClock, IconFire, IconSpan} from 'app/icons';
+import TimeSince from 'app/components/timeSince';
+
+import {CURRENT_LOCATION} from './types';
+
+enum EVENT_TYPE {
+  ERROR = 'error',
+  TRANSACTION = 'transaction',
+}
+
+type Props = {
+  relatedEvents: Array<TableDataRow>;
+  eventView: EventView;
+  organization: Organization;
+  currentLocation: CURRENT_LOCATION;
+  location: Location;
+};
+
+// List events that have the same tracing ID as the current Event
+const RelatedEvents = ({
+  currentLocation,
+  eventView,
+  relatedEvents,
+  organization,
+  location,
+}: Props) => {
+  const orgSlug = organization.slug;
+
+  const getTransactionLink = (transactionName: string, eventSlug: string) => {
+    return getTransactionDetailsUrl(
+      organization,
+      eventSlug,
+      transactionName,
+      location.query
+    );
+  };
+
+  const getEventTarget = (dataRow: TableDataRow & {type: EVENT_TYPE}) => {
+    if (currentLocation === CURRENT_LOCATION.DISCOVER) {
+      const eventSlug = generateEventSlug(dataRow);
+
+      return eventDetailsRouteWithEventView({
+        orgSlug,
+        eventSlug,
+        eventView,
+      });
+    }
+
+    if (dataRow.type === EVENT_TYPE.ERROR) {
+      return `/organizations/${orgSlug}/issues/${dataRow['issue.id']}/events/${dataRow.id}/`;
+    }
+
+    const eventSlug = generateEventSlug(dataRow);
+
+    return getTransactionLink(String(dataRow.title), eventSlug);
+  };
+
+  const renderEventId = (dataRow: TableDataRow & {type: EVENT_TYPE}) => (
+    <Tooltip title={t('View Event')}>
+      <StyledLink to={getEventTarget(dataRow)}>
+        {dataRow.id as React.ReactNode}
+      </StyledLink>
+    </Tooltip>
+  );
+
+  const renderIcon = (type: EVENT_TYPE) => {
+    if (type === EVENT_TYPE.ERROR) {
```

Comment:
There are other event types that *could* show up. CSP or default events are two other potentials. You might want to invert this logic and if the event type is transaction return the span icon, otherwise return fire.